### PR TITLE
ONNX Export All Cases of Softmax

### PR DESCRIPTION
**Pull request by lara-hdr**

### CODE REVIEW

**Comment by lara-hdr** [comment]
@houseroad, @zrphercule, could you please take a look a this

**Comment by houseroad** [comment]
Yeah, feel free to ping me :-)

**Review by houseroad — DISMISSED** [review]
Can we also add test cases for dtype argument?

File: test/onnx/test_pytorch_onnx_caffe2.py [hunk 1]
```diff
@@ -882,13 +882,29 @@ def forward(self, x):
             self.run_model_test(MyModel(), train=False, input=(x), batch_size=BATCH_SIZE, use_gpu=False)
 
     # TODO: Add test cases for prod once Caffe2 has support for ReduceProd
-    def test_softmax(self):
+    def test_softmax_dim(self):
         for i in range(7)[2:]:
```
Comment (houseroad) [inline test/onnx/test_pytorch_onnx_caffe2.py]: can we change this to range(2, 8)?

File: test/onnx/test_pytorch_onnx_caffe2.py [hunk 2]
```diff
@@ -882,13 +882,29 @@ def forward(self, x):
...
             model = nn.Softmax(dim=i - 1)
             dims = [2] * (i - 2) + [3, 4]
             input = torch.ones(*dims, requires_grad=True)
             self.run_model_test(model, train=False, batch_size=BATCH_SIZE, input=input)
 
+    def test_softmax(self):
```
Comment (houseroad) [inline test/onnx/test_pytorch_onnx_caffe2.py]: can we merge this test with the previous test (i.e., test_softmax_dim) or at least follow the style in the previous test. we can test the case which dim and axis have different semantics more thoroughly.

**Review by houseroad — COMMENTED** [review]
Thanks for updating the PR. Please address the inline comments. Thanks!

File: test/onnx/test_pytorch_onnx_caffe2.py (hunk 2 above)
Comment (lara-hdr) [reply test/onnx/test_pytorch_onnx_caffe2.py]: I merged test_softmax_dim and test_softmax, but kept test_softmax_dtype separately, as we don't need to re-test each case with dtype

**Review by houseroad — APPROVED** [review]
